src/geneweaver/api/controller/species.py

```diff
@@ -0,0 +1,24 @@
+"""Endpoints related to species."""
+from typing import Optional
+
+from fastapi import APIRouter, Depends, Query
+from geneweaver.api import dependencies as deps
+from geneweaver.api.services import species as species_service
+from geneweaver.core.enum import GeneIdentifier
+from typing_extensions import Annotated
+
+router = APIRouter(prefix="/species", tags=["species"])
+
+
+@router.get("/")
+def get_species(
+    cursor: Optional[deps.Cursor] = Depends(deps.cursor),
+    taxonomy_id: Annotated[
+        Optional[int], Query(format="int64", minimum=0, maxiumum=9223372036854775807)
+    ] = None,
+    gene_id_type: Optional[GeneIdentifier] = None,
```

Comment:
I think `reference_gene_id_type` would be a better name. This really will just check on what the reference type is, not all available types.

Reply:
Changed